tt_metal/api/command_queue.hpp

```diff
@@ -0,0 +1,104 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "types.hpp"
+
+#include "tt_metal/impl/buffers/buffer.hpp"
+
+//==================================================
+//                COMMAND QUEUE OPERATIONS
+//==================================================
+
+
+namespace tt::tt_metal{
+namespace v1 {
+
+/**
+ * @brief Reads a buffer from the device.
+ *
+ * @param cq The command queue used to dispatch the command.
+ * @param buffer The device buffer to read from.
+ * @param dst Pointer to the destination memory where data will be stored.
+ * @param blocking Indicates whether the operation is blocking.
+ */
+void EnqueueReadBuffer(
+    CommandQueue cq,
+    Buffer buffer,
+    void *dst,
+    bool blocking);
+
+/**
+ * @brief Writes data to a buffer on the device.
+ *
+ * @param cq The command queue used to dispatch the command.
+ * @param buffer The device buffer to write to.
+ * @param src Source data vector to write to the device.
+ * @param blocking Indicates whether the operation is blocking.
+ */
+void EnqueueWriteBuffer(
+    CommandQueue cq,
+    Buffer buffer,
+    const std::vector<uint32_t> &src,
+    bool blocking);
+
+/**
+ * @brief Writes data to a buffer on the device.
+ *
+ * @param cq The command queue used to dispatch the command.
+ * @param buffer The device buffer to write to.
+ * @param src Source data of type HostDataType to write to the device.
+ * @param blocking Indicates whether the operation is blocking.
+ */
+void EnqueueWriteBuffer(
+    CommandQueue cq,
+    Buffer buffer,
+    HostDataType src,
```

Comment:
Is this the "fix" for requiring a vector above?  Seems we should deprecate the vector version w/ 1.0 (tag it v0)?

Reply:
Changing to const std::byte *src